### DISCUSSION TITLE
Can't get Statamic to write contents

### DISCUSSION THREAD

**Discussion by hansipete** [deployphp/deployer#3270] [Help needed] [answered]
Hello hello, 

since a few days I'm struggling to get Statamic to work. I'm deploying to a Hetzner VPS (smallest for now, CX11 1vCPU, 2Gb RAM).

I set up the server running `dep provision -o remote_user=root`. Now I can deploy the site with `dep deploy`. My deploy.php looks something like below.

The site itself works, but I can not edit any content from the backend. I get the message: `file_put_contents('/home/deployer/xxxyyyzzz.de/releases/11/content/collections/pages/home.md): Failed to open stream: Permission denied`

I don't understand why this is, because there seems to be a symlink in that folder: `content -> ../../shared/content`

Hope somebody can help. 

Thank you,
Hans

```
<?php
namespace Deployer;

require 'recipe/statamic.php';

// get inspiration from here
// https://github.com/deployphp/deployer/issues/572#issuecomment-193311928

// Config

set('repository', 'git@github.com:XXX/YYY.git');
set('keep_releases', 3);

add('shared_files', ['.env']);

add('shared_dirs', [
    'content',
    'public/assets',
    'public/img',
    'public/storage',
]);
add('writable_dirs', [
    'content',
    'public/assets',
    'public/img',
    'public/storage',
    'bootstrap/cache']);

// Tasks
task('upload_assets', function () {
    upload(__DIR__ . "/public/assets/", '{{deploy_path}}/shared/public/assets/');
})->desc('upload all assets from localhost to server');

task('upload_env', function() {
    upload(__DIR__ . "/.env.staging", '{{deploy_path}}/shared/.env');
})->desc('Copy environment file');

task('npm:build', function() {
    run('cd {{release_or_current_path}} && npm ci && npm run production');
})->desc('npm run production');

// Hosts
host('xxxyyyzzz.de')
    ->set('remote_user', 'deployer')
    ->set('deploy_path', '~/xxxyyyzzz.de')
    ->set('branch', 'production');

// Hooks
after('deploy:failed', 'deploy:unlock');
```

**Comment by hansipete** [comment, marked as answer]
I am still confused how to get everything working, but made some progress.

This is my current `deployer.php` settings for files/folders:
```php
add('shared_dirs', [
    'storage',
    'content'
]);

add('shared_files', [
    '.env'
]);

add('writable_dirs', [
    'storage',
    'content',
    'public', 
    'bootstrap/cache',
]);
```

What did the trick for me was changing php-fpm user to `deployer`. This config can be done in `/etc/php/8.1/fpm/pool.d/www.conf`. Restarted fpm process and it was working (create, edit content, upload assets tested)

Another little adjustment needed in deployer.php: `host(...)→set('http_user', 'deployer')`

Next up I'm searching for a way to allow Statamic Git Integration to commit/push changes on the live server to the repository...

**Comment by rabol** [comment]
@hansipete  how did you configure the deployment script to keep your assets after a new deployment ?

I'm loosing all my images
